src/devices/machine/upd765.cpp

```diff
@@ -399,9 +406,11 @@ uint8_t ps2_fdc_device::sra_r()
 	floppy_info &fi = flopi[dor & 3];
 	if(fi.dir)
 		sra |= 0x01;
+	if(fi.dev && !fi.dev->wpt_r())
+		sra |= 0x02;
 	if(fi.index)
 		sra |= 0x04;
-	if(cur_rate >= 500000)
+	if(fi.dev && fi.dev->ss_r())
```

Comment:
Is this applicable to all variants?  I would assume there was some logic behind the previous implementation.